util/device_set.h

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provide generic device sets support for tests that require multiple
+//  devices to run
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_UTIL_DEVICE_SET_H
+#define __SYCLCTS_UTIL_DEVICE_SET_H
+
+#include "../tests/common/common.h"
+#include "kernel_restrictions.h"
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace sycl_cts::util {
+
+/** @brief Provides toolkit for working with a set of devices
+ *  @details For example, to have a greedy tests using all devices with and
+ *           without specific aspects we can:
+ *               const auto platforms = sycl::platform::get_platforms();
+ *               for (const auto& platform: platforms) {
+ *                 sycl::context ctx(platform.get_devices());
+ *
+ *                 auto devs_all = device_set(ctx);
+ *                 auto devs_fp16 = device_set::filtered(devs_all,
+ *                                                       sycl::aspect::fp16);
+ *                 auto devs_fp64 = device_set::filtered(devs_all,
+ *                                                       sycl::aspect::fp64);
+ *                 auto devs_core = devs_all;
+ *                 devs_core.removeDevsWith(sycl::aspect::fp16);
+ *                 devs_core.substract(devs_fp64);
+ *               }
+ */
+class device_set {
+  using StorageType = std::unordered_set<sycl::device>;
```

Comment:
@bader  should we use some container, instead of `std::unordered_set` or this container will be supported later in the hipSYCL?

Reply:
I think @illuhad should be able to answer this.
On your side, you can check if other containers not requiring `std::hash` specialization can be used instead of `std::unordered_set` while hipSYCL is not fixed.

Reply:
> you can check if other containers not requiring std::hash specialization can be used instead of std::unordered_set

@bader `std::set` can't be use, but std::vector can be used as a temporary solution and `std::unordered_set` will be hidden by a macro

But I suggest wait Aksel opinion

Reply:
@vasilytric, according to Aksel https://github.com/illuhad/hipSYCL/issues/680 is going to be fixed by the end of next week.

Reply:
@bader Thanks for the information. 
What we are going to do? Waiting for the fix or make a temporary solution by using `std::vector`?

Reply:
I suggest we wait for updated hipSYCL compiler.

Reply:
@bader Yes, let's wait

Reply:
See also internal https://gitlab.khronos.org/sycl/Specification/-/issues/586